ffi/MD5.lua

```diff
@@ -220,14 +228,16 @@ md5_proto.__index = md5_proto
 --- Feed content to md5 hashing instance.
 ---- @string luastr stream to process
 function md5_proto:update(luastr)
-    MD5Update(self.ctx, ffi.cast("const char*", luastr), #luastr)
+    local len = #luastr
+    local p = ffi.cast("const unsigned char *", luastr)
+    MD5Update(self.ctx, p, len)
 end
 
 --- Calculate md5 sum with the state of a md5 hashing instance.
 ---- @string luastr stream to process
 ---- @treturn string md5 sum in Lua string
 function md5_proto:sum(luastr)
-    local buf = ffi.new("char[33]")
+    local buf = ffi.new("unsigned char[33]")
```

Comment:
Doubtful that it would affect anything, I actually quite explicitly played around with this very change because I wondered if putting in an `unsigned` might help.

Reply:
It's unsigned on arm anyway ;).

(And mostly always explicitly an uint8_t in upstream's examples).

Reply:
Right, I'm just saying I did a few x86_64 experiments in Docker none of which made a difference on Clang-built LuaJIT. ;-) Regardless of cross-platform compatibility/best practices/etc., just to see if I could isolate something, anything at all.

Reply:
It's still especially confusing to me that locally it's a toss up or possibly as low as 1 in 4 that the problem occurs, while in the CI it's assured to go wrong.

Reply:
Yeah, I think I sometimes saw only 2 of the tests failing instead of the three during last night's runs. :?

Reply:
Oh, it does? Huh, I've always seen all three on CircleCI.